src/vm.rs

```diff
@@ -518,8 +519,11 @@ impl<'a, S: Storage, C: PevmChain> Vm<'a, S, C> {
         // Execute
         let mut db = match VmDb::new(self, tx_version.tx_idx, tx, from_hash, to_hash) {
             Ok(db) => db,
-            // TODO: Handle different errors differently
-            Err(_) => return VmExecutionResult::FallbackToSequential,
+            Err(err) => {
+                return VmExecutionResult::Err(VmExecutionError::ExecutionError(
```

Comment:
Hey @hai-rise, I'm not the biggest fan of the stuttering and verbosity here. 

An alternative could impl method e.g.

```
impl VmExecutionError {
    pub fn execution_error(error: ExecutionError) -> Self {
        VmExecutionError::ExecutionError(error)
    }
}
```
Please lmk what you think 🙏

Reply:
I agree! I think a clean solution is to:
- Bring `ReadError` from `lib.rs` here as a `pub(crate)` (as a related chore).
- Implement a `From<ReadError>`  for `VmExecutionError` as the source-of-truth for read error handling behaviour.
```rust
impl From<ReadError> for VmExecutionError {
    fn from(err: ReadError) -> Self {
        match err {
            ReadError::InconsistentRead => VmExecutionError::Retry,
            ReadError::SelfDestructedAccount => VmExecutionError::FallbackToSequential,
            ReadError::Blocking(tx_idx) => VmExecutionError::Blocking(tx_idx),
            _ => VmExecutionError::ExecutionError(EVMError::Database(err)),
        }
    }
}
```
- Then we can simplify this instance to:
```rust
let mut db = VmDb::new(self, tx_version.tx_idx, tx, from_hash, to_hash)
            .map_err(VmExecutionError::from)?;
```
- And the several match arms below to:
```rust
Err(EVMError::Database(read_error)) => Err(read_error.into()),
```

Also, note that the current change is a semantic change (from returning `FallbackToSequential` to returning `ExecutionError`) that fails test.